Paper: Extension of the OpenMC depletion module for transport-independent depletion (#1113)
scipy-conference/scipy_proceedings: papers/oleksandr_yardas/main.md

    the higher floating point errors, however their magnitudes are marginal. 

The general trend in our results is that concentration errors are smaller
for shorter depletion time steps, and larger for longer depletion time
steps, however some nuclides exhibit more complicated behavior. 

::::{figure}
:label: actinide-error-days

:::{list-table}
- - (fig:actinides-error-constant-xs-days)=
    ![ ](figs/actinides_constant_xs_predictor_fission_q_days.*)
  - (fig:actinides-error-updating-xs-days)=
    ![ ](figs/actinides_updating_xs_predictor_fission_q_days.*)
:::
Relative actinide concentration error using 3-day time steps at 3, 12, 21, and
30 days of depletion for [({subEnumerator})](fig:actinides-error-constant-xs-days) constant
cross sections; [({subEnumerator})](fig:actinides-error-updating-xs-days) updating cross
sections.
::::

[](fig:actinides-error-constant-xs-days) and
[](fig:actinides-error-updating-xs-days) show the relative error in predicted
actinide concentration for constant cross sections and updating cross sections,
respectively, using 3-day time steps. [](fig:actinides-error-constant-xs-months)

> and [](fig:actinides-error-updating-xs-months) show the same respective

Reviewer: ```suggestion
30 days of depletion for [({subEnumerator})](#fig:actinides-error-constant-xs-days) constant
cross sections; [({subEnumerator})](#fig:actinides-error-updating-xs-days) updating cross
sections.
::::

[](#fig:actinides-error-constant-xs-days) and
[](#fig:actinides-error-updating-xs-days) show the relative error in predicted
actinide concentration for constant cross sections and updating cross sections,
respectively, using 3-day time steps. [](#fig:actinides-error-constant-xs-months)
and [](#fig:actinides-error-updating-xs-months) show the same respective
```

errors, (5\% or less), whereas the less abundant actinides (isotopes of Am,
isotopes of Cm, $^{241}$Pu, $^{242}$Pu) tend to have high (more
than 10\%) concentration errors depending on the nuclide of interest.

::::{figure}
:label: actinide-error-months

::::{list-table}
- - (fig:actinides-error-constant-xs-months)=
    ![  ](figs/actinides_constant_xs_predictor_fission_q_months.*)
  - (fig:actinides-error-updating-xs-months)=
     ![ ](figs/actinides_updating_xs_predictor_fission_q_months.*)
:::
Relative actinide concentration error using 30-day time steps at 30, 120, 210, and
300 days of depletion for [({subEnumerator})](fig:actinides-error-constant-xs-months) constant
cross sections; [({subEnumerator})](fig:actinides-error-updating-xs-months) updating cross
sections.
::::

[](fig:fp-error-constant-xs-days) and [](fig:fp-error-updating-xs-days) show the
relative error in predicted fission product concentration using constant cross
sections and updating cross sections, respectively, using 3-day time steps.
Figures [](fig:fp-error-constant-xs-months) and
[](fig:fp-error-updating-xs-months) show the same respective quantities for

> 30-day time steps. Similar to the actinides, updating the cross sections at each

Reviewer: ```suggestion
300 days of depletion for [({subEnumerator})](#fig:actinides-error-constant-xs-months) constant
cross sections; [({subEnumerator})](#fig:actinides-error-updating-xs-months) updating cross
sections.
::::

[](#fig:fp-error-constant-xs-days) and [](#fig:fp-error-updating-xs-days) show the
relative error in predicted fission product concentration using constant cross
sections and updating cross sections, respectively, using 3-day time steps.
[](#fig:fp-error-constant-xs-months) and
[](#fig:fp-error-updating-xs-months) show the same respective quantities for
30-day time steps. Similar to the actinides, updating the cross sections at each
```

concentration errors are orders of magnitude less abundant in the fuel than
$^{235}$U, so their effect on fission product concentration error is
proportionally small.

Concentration errors for low-abundance nuclides using 30-day time steps follow
an interesting trend where there is an initial large negative concentration
error that becomes more positive over time. This behavior is due to
the static cross sections and fluxes producing the same amount of these
nuclides every timestep, whereas in the transport-coupled case,
net production of the low-abundance nuclides decreases over time.

::::{figure}
:label: fp-error-days

:::{list-table}
- - (fig:fp-error-constant-xs-days)=
     ![  ](figs/fission_products_constant_xs_predictor_fission_q_days.*)
  - (fig:fp-error-updating-xs-days)=
    ![ ](figs/fission_products_updating_xs_predictor_fission_q_days.*)
:::
Relative fission produt concentration error using 3-day time steps at 3, 12, 21, and
30 days of depletion for [({subEnumerator})](fig:fp-error-constant-xs-days) constant
cross sections; [({subEnumerator})](fig:fp-error-updating-xs-days) updating cross

> sections.

Reviewer: ```suggestion
30 days of depletion for [({subEnumerator})](#fig:fp-error-constant-xs-days) constant
cross sections; [({subEnumerator})](#fig:fp-error-updating-xs-days) updating cross
sections.
```

::::

::::{figure}
:label: fp-error-months

:::{list-table}
- - (fig:fp-error-constant-xs-months)=
    ![  ](figs/fission_products_constant_xs_predictor_fission_q_months.*)
  - (fig:fp-error-updating-xs-months)=
    ![ ](figs/fission_products_updating_xs_predictor_fission_q_months.*)
:::
Relative fission produt concentration error using 30-day time steps at 30, 120, 210, and
300 days of depletion for [({subEnumerator})](fig:fp-error-constant-xs-months) constant
cross sections; [({subEnumerator})](fig:fp-error-updating-xs-months) updating cross
sections.
::::

Repeating this analysis for both the CASMO-8 and CASMO-40 multi-group
structures did not yield noticeable decreases in nuclide concentration
errors for transport-independent depletion over the one-group case.  Figures
[](fig:actinides-error-casmo8-constant-xs-days) and
[](fig:actinides-error-casmo40-xs-days) show the relative error in
predicted actinide concentration using the CASMO-8 and CASMO-40 group
structures, respectively, using 3-day time steps. Figures
[](fig:actinides-error-casmo8-constatn-xs-months) and

> [](fig:actinides-error-casmo40-xs-months) show the same respective

Reviewer: ```suggestion
300 days of depletion for [({subEnumerator})](#fig:fp-error-constant-xs-months) constant
cross sections; [({subEnumerator})](#fig:fp-error-updating-xs-months) updating cross
sections.
::::

Repeating this analysis for both the CASMO-8 and CASMO-40 multi-group
structures did not yield noticeable decreases in nuclide concentration
errors for transport-independent depletion over the one-group case.
[](#fig:actinides-error-casmo8-constant-xs-days) and
[](#fig:actinides-error-casmo40-xs-days) show the relative error in
predicted actinide concentration using the CASMO-8 and CASMO-40 group
structures, respectively, using 3-day time steps.
[](#fig:actinides-error-casmo8-constatn-xs-months) and
[](#fig:actinides-error-casmo40-xs-months) show the same respective
```

important.

::::{figure}
:label: actinide-error-casmo-days

:::{list-table}
- - (fig:actinides-error-casmo8-constant-xs-days)=
    ![  ](figs/actinides_casmo8_constant_xs_predictor_fission_q_days.*)
  - (fig:actinides-error-casmo40-constant-xs-days)=
    ![ ](figs/actinides_casmo40_constant_xs_predictor_fission_q_days.*)
:::
Relative actinide concentration error using constant cross sections and 3-day
time steps at 3, 12, 21, and 30 days of depletion for the
[({subEnumerator})](fig:actinides-error-casmo8-constant-xs-days) CASMO-8 group
structure; [({subEnumerator})](fig:actinides-error-casmo40-constant-xs-days)
CASMO-40 group structure.
::::

::::{figure}
:label: actinide-error-casmo-months

:::{list-table}
- - (fig:actinides-error-casmo8-constant-xs-months)=
    ![  ](figs/actinides_casmo8_constant_xs_predictor_fission_q_months.*)
 -  (fig:actinides-error-casmo40-constant-xs-months)=
    ![ ](figs/actinides_casmo40_constant_xs_predictor_fission_q_months.*)
:::
Relative actinide concentration error using constant cross sections and 3-day
time steps at 30, 120, 210, and 300 days of depletion for the
[({subEnumerator})](fig:actinides-error-casmo8-constant-xs-months) CASMO-8 group

> structure; [({subEnumerator})](fig:actinides-error-casmo40-constant-xs-months)

Reviewer: ```suggestion
[({subEnumerator})](#fig:actinides-error-casmo8-constant-xs-days) CASMO-8 group
structure; [({subEnumerator})](#fig:actinides-error-casmo40-constant-xs-days)
CASMO-40 group structure.
::::

::::{figure}
:label: actinide-error-casmo-months

:::{list-table}
- - (fig:actinides-error-casmo8-constant-xs-months)=
    ![  ](figs/actinides_casmo8_constant_xs_predictor_fission_q_months.*)
 -  (fig:actinides-error-casmo40-constant-xs-months)=
    ![ ](figs/actinides_casmo40_constant_xs_predictor_fission_q_months.*)
:::
Relative actinide concentration error using constant cross sections and 3-day
time steps at 30, 120, 210, and 300 days of depletion for the
[({subEnumerator})](#fig:actinides-error-casmo8-constant-xs-months) CASMO-8 group
structure; [({subEnumerator})](#fig:actinides-error-casmo40-constant-xs-months)
```

The first case serves as a reference solution we use to estimate the error
in the solution of the new method. We used 25 inactive and 125 active
batches, with $10^6$ particles per batch. For the second case, we used three
different group structures to test the effect on solution accuracy: one
group, CASMO-8, and CASMO-40. The third case is much slower than the first
case as the cross sections need to be reloaded after each depletion step.
The third case is a sanity check that the cross sections are being computed
correctly, as it should have a smaller error than the second case.
Due to the cross section discretization that occurs before performing
depletion, we do not expect the third case results to converge to the first
case results in limit of infinite particles, but we would expect it to
converge in the limit of infinite particles _and_ energy groups. The
third case was only run for one-group transport-independent depletion. 

[](#tab:mat-params) and [](#tab:mat-comps) contain the material parameters and
compositions of our model, and [](#tab:geo-params) contains the geometric
parameters.  We used the ENDF/B-VII.1 nuclear data library available at

> openmc.org/official-data-libraries. We used the ENDF/B-VII.1 depletion

Reviewer: ```suggestion
[openmc.org/data](https://openmc.org/data/#endf-b-vii-1). We used the ENDF/B-VII.1 depletion
```

> chain in the PWR spectrum available at openmc.org/depletion-chains.

Reviewer: ```suggestion
chain in the PWR spectrum also available at [openmc.org/data](https://openmc.org/data/#endf-b-vii-1-chain-thermal-spectrum).
```
